connector/countconnector/connector.go

```diff
@@ -0,0 +1,125 @@
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (c *count) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	countMetrics := pmetric.NewMetrics()
+	for i := 0; i < td.ResourceSpans().Len(); i++ {
+		resourceSpan := td.ResourceSpans().At(i)
+		countResource := countMetrics.ResourceMetrics().AppendEmpty()
+		resourceSpan.Resource().Attributes().CopyTo(countResource.Resource().Attributes())
+
+		countScope := countResource.ScopeMetrics().AppendEmpty()
+		countScope.Scope().SetName(scopeName)
+
+		count := 0
+		for j := 0; j < resourceSpan.ScopeSpans().Len(); j++ {
+			count += resourceSpan.ScopeSpans().At(j).Spans().Len()
+		}
+		setCountMetric(countScope.Metrics().AppendEmpty(), c.Config.Traces, count)
+	}
+	return c.Metrics.ConsumeMetrics(ctx, countMetrics)
+}
+
+func (c *count) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	countMetrics := pmetric.NewMetrics()
+	for i := 0; i < md.ResourceMetrics().Len(); i++ {
+		resourceMetric := md.ResourceMetrics().At(i)
+		countResource := countMetrics.ResourceMetrics().AppendEmpty()
+		resourceMetric.Resource().Attributes().CopyTo(countResource.Resource().Attributes())
+
+		countScope := countResource.ScopeMetrics().AppendEmpty()
+		countScope.Scope().SetName(scopeName)
+
+		count := 0
+		for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
+			countMetrics := resourceMetric.ScopeMetrics().At(j).Metrics()
+			for k := 0; k < countMetrics.Len(); k++ {
+				countMetric := countMetrics.At(k)
+				switch countMetric.Type() {
+				case pmetric.MetricTypeGauge:
+					count += countMetric.Gauge().DataPoints().Len()
+				case pmetric.MetricTypeSum:
+					count += countMetric.Sum().DataPoints().Len()
+				case pmetric.MetricTypeSummary:
+					count += countMetric.Summary().DataPoints().Len()
+				case pmetric.MetricTypeHistogram:
+					count += countMetric.Histogram().DataPoints().Len()
+				case pmetric.MetricTypeExponentialHistogram:
+					count += countMetric.ExponentialHistogram().DataPoints().Len()
+				}
+			}
+		}
+		setCountMetric(countScope.Metrics().AppendEmpty(), c.Config.Metrics, count)
+	}
+	return c.Metrics.ConsumeMetrics(ctx, countMetrics)
+}
+
+func (c *count) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	countMetrics := pmetric.NewMetrics()
+	for i := 0; i < ld.ResourceLogs().Len(); i++ {
+		resourceLog := ld.ResourceLogs().At(i)
+		countResource := countMetrics.ResourceMetrics().AppendEmpty()
+		resourceLog.Resource().Attributes().CopyTo(countResource.Resource().Attributes())
+
+		countScope := countResource.ScopeMetrics().AppendEmpty()
+		countScope.Scope().SetName(scopeName)
+
+		count := 0
+		for j := 0; j < resourceLog.ScopeLogs().Len(); j++ {
+			count += resourceLog.ScopeLogs().At(j).LogRecords().Len()
+		}
+		setCountMetric(countScope.Metrics().AppendEmpty(), c.Config.Logs, count)
+	}
+	return c.Metrics.ConsumeMetrics(ctx, countMetrics)
+}
+
+func setCountMetric(countMetric pmetric.Metric, metricType TypeConfig, count int) {
```

Comment:
Can accept int64(uint64) for count to not (very very very unlikely/probably impossible) overflow on a 32bit system.